Jenkinsfile

```diff
@@ -256,18 +300,36 @@ void runIntegrationTest(String phpVersion, String client, String psrImplem, Stri
     node('docker') {
         deleteDir()
         try {
-            sh "docker run --name mysql -e MYSQL_ROOT_PASSWORD=root -e MYSQL_USER=akeneo_pim -e MYSQL_PASSWORD=akeneo_pim -e MYSQL_DATABASE=akeneo_pim -d mysql:5.5 --sql-mode=ERROR_FOR_DIVISION_BY_ZERO,NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"
-
             dir('pim') {
                 unstash "pim_community_dev_${pimVersion}"
-                sh "docker run --name akeneo-pim --link mysql:mysql -v \$(pwd):/home/docker/pim -d carcel/akeneo-apache:php-5.6"
-                sh "docker exec akeneo-pim pim/app/console pim:install -e prod"
+
+                if(pimVersion.equals("master")){
+                    sh "docker run --name mysql -e MYSQL_ROOT_PASSWORD=root -e MYSQL_USER=akeneo_pim -e MYSQL_PASSWORD=akeneo_pim -e MYSQL_DATABASE=akeneo_pim -d mysql:5.7"
+                    sh "docker run --name elasticsearch -e ES_JAVA_OPTS=\"-Xms256m -Xmx256m\" -d elasticsearch:5"
+                    sh "docker run --name akeneo-pim --link mysql:mysql --link elasticsearch:elasticsearch -v \$(pwd):/srv/pim -w /srv/pim -d akeneo/fpm:php-7.1"
+                    sh "docker run --name httpd --link akeneo-pim:fpm -v \$(pwd):/srv/pim -v \$(pwd)/docker/httpd.conf:/usr/local/apache2/conf/httpd.conf -v \$(pwd)/docker/akeneo.conf:/usr/local/apache2/conf/vhost.conf -w /srv/pim -d httpd:2.4"
+                } else {
+                    sh "docker run --name mysql -e MYSQL_ROOT_PASSWORD=root -e MYSQL_USER=akeneo_pim -e MYSQL_PASSWORD=akeneo_pim -e MYSQL_DATABASE=akeneo_pim -d mysql:5.5"
+                    sh "docker run --name akeneo-pim --link mysql:mysql -v \$(pwd):/srv/pim -v \$(pwd)/docker/akeneo.conf:/etc/apache2/sites-available/000-default.conf:ro -w /srv/pim -d akeneo/apache-php:php-5.6"
+                }
+
+                // Wait for mysql container ready
+                sh "sleep 10"
+                sh "docker exec akeneo-pim app/console pim:install -e prod"
             }
 
-            docker.image("carcel/php:${phpVersion}").inside("--link akeneo-pim:akeneo-pim -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/usr/bin/docker") {
-                unstash "php-api-client_${client}_${psrImplem}_php-${phpVersion}".replaceAll("/", "_")
-                sh "mkdir -p build/logs/"
-                sh "sudo ./bin/phpunit -c phpunit.xml.dist --log-junit build/logs/phpunit_integration.xml"
+            unstash "php-api-client_${client}_${psrImplem}_php-${phpVersion}".replaceAll("/", "_")
+            sh "mkdir -p build/logs/"
+
+            if(pimVersion.equals("master")){
+                docker.image("akeneo/php:${phpVersion}").inside("--link akeneo-pim:akeneo-pim --link httpd:httpd -v /var/run/docker.sock:/var/run/docker.sock -v /usr/bin/docker:/usr/bin/docker -w /home/docker/client --privileged") {
+                    sh "sed -i \"s#baseUri: .*#baseUri: 'http://httpd'#g\" etc/parameters.yml"
```

Comment:
move it in checkout: it's the job of this function to prepare everything, it will avoid another `if else` here